website_oca_integrator/models/res_partner.py

```diff
@@ -0,0 +1,281 @@
+             item['parent_id_count']) for item in member_data)
+
+        for partner in self:
+            partner.member_count = member_mapped_data.get(partner.id, 0)
+
+    @api.multi
+    @api.depends('author_ids', 'author_ids.partner_id',
+                 'author_ids.module_qty')
+    def _compute_module_count(self):
+        for partner in self:
+            partner.module_count = sum(
+                [author.module_qty for author in partner.author_ids])
+
+    @api.multi
+    @api.depends('author_ids', 'is_integrator',
+                 'author_ids.module_ids.product_template_id')
+    def _compute_developed_modules(self):
+        for partner in self:
+            if partner.is_integrator:
+                module_list = self.env['odoo.module'].search(
+                    [('author_ids', 'in', partner.author_ids.ids)])
+                product_list = self.env['product.template'].search(
+                    [('odoo_module_id', 'in', module_list.ids)])
+                partner.developed_module_ids = [(6, 0, product_list.ids)]
+
+    @api.multi
+    def write(self, vals):
+        # clear github organization data if partner is not company
+        if not vals.get('is_company', True):
+            vals['github_organization'] = False
+        res = super(ResPartner, self).write(vals)
+        return res
+
+    def update_contributor_modules(self, contributor, modules):
+        """
+        Update contributor with fetched modules from Github.
+        """
+        github_product_templates = self.env['odoo.module'].search(
+            [('technical_name', 'in', list(modules.keys()))]).mapped(
+            'product_template_id')
+
+        contributor_module_lines = contributor.contributor_module_line_ids
+        product_mapped_data = dict([(x.product_template_id.id, x.id) for x in
+                                    contributor_module_lines])
+        current_product_templates = contributor_module_lines.mapped(
+            'product_template_id')
+
+        total_product_templates = github_product_templates \
+            | current_product_templates
+        common_product_templates = github_product_templates \
+            & current_product_templates
+        new_product_templates = github_product_templates \
+            - current_product_templates
+
+        update_records = [(1, product_mapped_data[x.id], {
+            'date_pr_merged': modules[x.odoo_module_id.technical_name]})
+            for x in common_product_templates]
+        new_records = [(0, 0,
+                        {'product_template_id': x.id,
+                         'date_pr_merged': modules[
+                             x.odoo_module_id.technical_name]})
+                       for x in new_product_templates]
+
+        if len(total_product_templates) <= 5:
+            contributor.write(
+                {'contributor_module_line_ids': update_records + new_records})
+        else:
+            remove_products_count = len(total_product_templates) - 5
```

Comment:
Let’s say, we got only 2 modules in 1st sync and 4 modules in second sync for contributor. In that case we only need to remove 1 module of the 1st sync as we need to maintain only 5 latest modules. If we do product.template(1,2)[:-5] then it will result an empty list. And hence, the 1 product of 1st sync will not be deleted.

Reply:
@st-aprajapati I see what you are saying now, but this entire algorithm `feels` clunky. @pedrobaeza could you take a look for me please as a second opinion?

Reply:
The simplest way is to always replace the content with `[(6, 0, <ids>)]` and you don't need to make this kind of things. The performance is going to be similar.

Reply:
@pedrobaeza I think, it will not work because this is one2many field.

Reply:
OK, got it. Then this is the best choice.

You can use `[(5, ), (0, 0, {...}), (0, 0, {...})]` for starting with a new list from scratch, but in this case you will recreate a lot of records.